### PR TITLE
Avoid potentially spilling credentials to state

### LINKED ISSUE #236: Custom providers can leak credentials to state file from environment variables
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

If using `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` as environment variables and using a custom (non-default) provider, the contents of these environment variables can be leaked into the state file in clear text.

### Steps to reproduce

1. Start with a program similar to this:
```
import * as pulumi from "@pulumi/pulumi";
import * as awsnative from "@pulumi/aws-native";

const prov = new awsnative.Provider("prov", {region: "us-west-2"});
// Create an AWS resource (S3 Bucket)
const bucket1 = new awsnative.s3.Bucket("my-bucket1", {}, {provider: prov});
const bucket2 = new awsnative.s3.Bucket("my-bucket2", {});

// Export the name of the bucket
export const bucket1Name = bucket1.id;
export const bucket2Name = bucket2.id;
```

2. Export the above environment variables with AWS credentials. `export AWS_SECRET_ACCESS_KEY=xxyyzz`
3. Run a `pulumi up` and dump the resulting state - `pulumi stack export | grep accessKey`
4. Notice `xxyyzz` in plain text.

### PR DESCRIPTION
**Pull request by viveklak**
Fixes #236

### CODE REVIEW

File: provider/pkg/schema/gen.go [hunk 1]
```diff
@@ -132,104 +135,9 @@ func GatherPackage(supportedResourceTypes []string, jsonSchemas []jsschema.Schem
 		Provider: pschema.ResourceSpec{
 			ObjectTypeSpec: pschema.ObjectTypeSpec{
 				Description: "The provider type for the AWS native package. By default, resources use package-wide configuration settings, however an explicit `Provider` instance may be created and passed during resource construction to achieve fine-grained programmatic control over provider settings. See the [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.",
-				Properties: map[string]pschema.PropertySpec{
```
Comment (viveklak) [inline provider/pkg/schema/gen.go:135]: I believe the Properties defined here are redundant given the input properties below.

Comment (mikhailshilkov) [reply provider/pkg/schema/gen.go:135]: `Properties` are outputs of the provider resource. Don't we define those in other provides?

Comment (viveklak) [reply provider/pkg/schema/gen.go:135]: Doesn't seem like it? See https://github.com/pulumi/pulumi-kubernetes/blob/master/provider/pkg/gen/schema.go#L95 and https://github.com/pulumi/pulumi-kubernetes/blob/master/provider/pkg/gen/schema.go#L95.

Comment (mikhailshilkov) [reply provider/pkg/schema/gen.go:135]: AWS does have them:
image

Comment (viveklak) [reply provider/pkg/schema/gen.go]: I removed the properties change for now. I did mark the 3 sensitive fields as secrets though.

**Comment by viveklak** [comment]
Adding @pgavlin to make sure we are being consistent with the spirit of https://github.com/pulumi/pulumi-azure/pull/583/ which was the last incarnation of this issue.

**Review by mikhailshilkov — APPROVED** [review]
I would be okay to exclude the secret properties from provider outputs - I don't think anyone would expect to get them as outputs

**Review by pgavlin — APPROVED** [review]

**Comment by viveklak** [comment]
> I would be okay to exclude the secret properties from provider outputs - I don't think anyone would expect to get them as outputs

Dropped the secret fields.

**Pull request merged**